avogadro/qtgui/meshgenerator.h

```diff
@@ -144,26 +220,24 @@ class AVOGADROQTGUI_EXPORT MeshGenerator : public QThread
   Vector3f m_stepSize;      /** The step size vector for cube. */
   Vector3f m_min;           /** The minimum point in the cube. */
   Vector3i m_dim;           /** The dimensions of the cube. */
-  Core::Array<Vector3f> m_vertices, m_normals;
-  Core::Array<unsigned int> m_indices;
+
+  Core::Array<Vector3f> points, normals;  
+  std::vector<gridEdge> gridEdges; // size (m_dim.y() * m_dim.z())
+  std::vector<unsigned char> cubeCases; //size ((m_dim.x() - 1) * (m_dim.y() - 1) * (m_dim.z() - 1))
+  std::vector<int> triCounter;  // size ((m_dim.y() - 1) * (m_dim.z() - 1))
+  std::vector<unsigned char> edgeCases; // size ((m_dim.x() - 1) * (m_dim.y()) * (m_dim.z()))
+  Core::Array<Vector3f> tris; // triangles of a mesh 
```

Comment:
Please remove any un-used variables (from Marching Cubes)

Please also use `m_triangles` (or similar) to keep consistent with member variables through the rest of the code.